[Cantina-Info-2]: Add predicate check to end of refinancing

PowVT opened the pull request:
During the Cantina audit it was noticed that the predicate check which is present in all other loan origination flows (initialize, rollover, migrations) was not present in the refinancing flow. To ensure protection from the borrower potentially manipulating collateral due to a safe transfer callback, specifically callback in the promissory notes `safeMint()` call it is important to add this check to the end of the refinancing flow as well.

To implement this change, the `runPredicatesCheck()` function was moved from the OriginationController to the OriginationConfiguration contract so it could be accessed by the refinancing contract as well. This should not increase the number of external calls from the OriginationController contract since it already had to make a call the OriginationConfiguration contract to check if the verifier in the predicates is whitelisted. Now the `runPredicatesCheck()` method can call the `isAllowedVerifier` natively and can also be used my contract which have the ORIGINATOR_ROLE in the lending protocol.

At the user level the implications of this, are they must define using the predicates array exactly what they want to be in the loan they are refinancing, similar to a rollover using a collection wide offer except the new lender does not need to sign anything.

kkennis commented:
Code looks good, but is `OriginationConfiguration` still a good name for the contract which handles predicates checks and other verification? It seems to have expanded in scope beyond configuration, since a predicates check doesn't rely on any configured values...maybe `OriginationHelpers`?

PowVT commented:
Agree with this comment, the scope of `OriginationConfiguration.sol` has now changed to include the validation functions and I have updated the contracts name to `OriginationHelpers.sol` to reflect this. Also updated all referenced throughout the codebase to refer to this contract as `originationHelpers`.
